### PR TITLE
feat: account selector

### PR DESCRIPTION
**Pull request by edgarkhanzadian**
Implement account selector with animations

https://github.com/user-attachments/assets/d9637c43-fcb4-4878-ba18-7720ebd812d2

### CODE REVIEW

File: apps/mobile/src/components/account-selector-sheet.tsx [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  return (
+    <>
+      <Animated.View style={placeholderUpAnimatedStyles} />
+      <GestureDetector gesture={gestures}>
+        <Animated.View
+          onLayout={e => {
+            const { height, width } = e.nativeEvent.layout;
+            thisItemDimensions.value = { height, width };
+          }}
+          // @ts-expect-error some ref type issues, it should be fine
```
Comment (pete-watters) [inline apps/mobile/src/components/account-selector-sheet.tsx:125]: 😰

**Review by pete-watters — APPROVED** [review]
🚀

File: apps/mobile/src/components/account-selector-sheet.tsx (hunk 1 above)
Comment (fbwoolf) [reply apps/mobile/src/components/account-selector-sheet.tsx:125]: Can this be explained? I'm not sure we typically do this? 🤔

File: apps/mobile/src/components/account-selector-sheet.tsx [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+import { useRouter } from 'expo-router';
+
+import { Box, IconProps } from '@leather.io/ui/native';
+
+import { getAvatarIcon } from './avatar-icon';
+import { Modal } from './bottom-sheet-modal';
+import { AccountCard } from './wallet-settings/account-card';
+
+const APPROXIMATE_DEFAULT_ITEM_SIZE = { height: 150, width: 300 };
+function DraggableAccount({
```
Comment (fbwoolf) [inline apps/mobile/src/components/account-selector-sheet.tsx:25]: Would it make sense to start pulling some of this code apart and using the `features` pattern? Some of these files are quite large and hard to follow. Would `account-selector-sheet` be it's own feature with local component files like `draggable-account`? In the extension, we typ try to only have one component per file if possible for readability.

Comment (fbwoolf) [reply apps/mobile/src/components/account-selector-sheet.tsx:25]: Also, will the `draggable` code be reused? 🤔 Maybe it needs to be more generic?

Comment (pete-watters) [reply apps/mobile/src/components/account-selector-sheet.tsx:25]: Good points 👍 I've been a bit trigger happy approving mobile stuff to increase velocity but as more of us are working on it now we should take extra care to increase the quality

**Review by fbwoolf — APPROVED** [review]

File: apps/mobile/src/features/account-selector-sheet.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            return (
+              <Draggable
+                idx={idx}
+                direction={direction}
+                scrollViewRef={scrollViewRef}
+                placeholderIdx={placeholderIdx}
+                accountsLength={accounts.length}
+                key={account.id}
+                accountId={account.id}
+                Component={() => (
```
Comment (fbwoolf) [inline apps/mobile/src/features/account-selector-sheet.tsx:76]: Curious on this bc I keep seeing it, we never have captialized props in the extension (even if we are passing ReactNodes), is that typ in RN code? Not opposed to it, but I think in the icon work I refactored some to lowercase `icon` out of habit.

File: apps/mobile/src/components/account-selector-sheet.tsx (hunk 2 above)
Comment (fbwoolf) [reply apps/mobile/src/components/account-selector-sheet.tsx]: Lgtm, thanks for doing! 👍

File: apps/mobile/src/features/account-selector-sheet.tsx (hunk 3 above)
Comment (pete-watters) [reply apps/mobile/src/features/account-selector-sheet.tsx:76]: This looks weird to me also, but perhaps is it just when the prop is `Component`? 

There is a convention I'm familiar with where components are capitalised to indicate that e.g. `SomeComponent.tsx`

File: apps/mobile/src/features/account-selector-sheet.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        idx1 === null ||
+        idx2 === null ||
+        !checkIdxWithinBounds(idx1) ||
+        !checkIdxWithinBounds(idx2)
+      )
+        return;
+      const temp = accountIds[idx1];
+      /** @ts-expect-error we know that both indexes have both value in accounts array */
+      accountIds[idx1] = accountIds[idx2];
+      /** @ts-expect-error we know that both indexes have both value in accounts array */
```
Comment (kyranjamie) [inline apps/mobile/src/features/account-selector-sheet.tsx:44]: Any way we can avoid doing this with some runtime checks? Would say ts-expect-error isn't something we want to rely on too much

File: apps/mobile/src/features/account-selector-sheet.tsx (hunk 3 above)
Comment (kyranjamie) [reply apps/mobile/src/features/account-selector-sheet.tsx:76]: Why pass `() => ReactNode` instead of `ReactNode`?

File: apps/mobile/src/components/draggable.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  placeholderIdx,
+  scrollViewRef,
+  direction,
+  onAccountPress,
+  swapAccountIndexes,
+  Component,
+}: {
+  idx: number;
+  accountId: string;
+  accountsLength: number;
```
Comment (kyranjamie) [inline apps/mobile/src/components/draggable.tsx:26]: This component being called `Draggable` makes me think it's a generic component that can be used elsewhere. But it can't, it's specific to the account re-ordering use-case.

Suggest either refactoring to be reusable, or renaming and relocating to the feature folder

Comment (fbwoolf) [reply apps/mobile/src/components/draggable.tsx:26]: I was trying to get him to refactor it so it was reusable, but I overlooked there were still account specific prop names. My thought was we will have other uses for the draggable code and it should be generic?

File: apps/mobile/src/components/wallet-settings/account-card.tsx [hunk 6]
```diff
@@ -7,10 +9,14 @@ export function AccountCard({
...
   name,
   onPress,
+  onLongPress,
+  onLayout,
 }: {
   Icon: React.FC<IconProps>;
   name: string;
   onPress?(): unknown;
+  onLongPress?(): unknown;
+  onLayout?(e: LayoutChangeEvent): void;
```
Comment (kyranjamie) [inline apps/mobile/src/components/wallet-settings/account-card.tsx:19]: Not sure why mixing void and unknown. If no return value expected, `void` should be used imo, as done in docs https://www.typescriptlang.org/docs/handbook/2/functions.html
